src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -12322,7 +12322,12 @@ def resample(
 
         validate_resample_supported_by_snowflake(resample_kwargs)
 
-        frame = self._modin_frame
+        axis = resample_kwargs.get("axis", 0)
+        rule = resample_kwargs.get("rule")
+        on = resample_kwargs.get("on")
+
+        # Supplying 'on' to Resampler replaces the existing index of the DataFrame with the 'on' column
+        frame = self.set_index(on)._modin_frame if on else self._modin_frame
```

Comment:
This weird edge case might happen here if on label has "Falsy" value, for example `0`.
Can you please replace this with 
`frame = self.set_index(on)._modin_frame if on is not None else self._modin_frame`

Maybe also add this test case for completeness.

Reply:
`on` is guaranteed to be a string so I don't think this case gets hit but added it to be safe.